Proxy: Simplify build system

jwendell opened the pull request.

jwendell committed:
Simplify build system

dgn commented on istio-proxy/istio-proxy.spec:
```diff
@@ -40,44 +39,62 @@ BuildRequires:  gcc
-The Istio Proxy is a microservice proxy that can be used on the client and server side, and forms a microservice mesh. The Proxy supports a large number of features.
+%build
+# BEGIN GN FIXME: Move GN to its own package?
+pushd gn
```
not sure where this is coming from? it's not downloaded by `update.sh`

jwendell replied:
This is needed for building multi-arch; I brought it here from product in a attempt to have equal .specs. I've just removed it as we are not building multi-arch for community (copr does not even support the other arches).

dgn commented on istio-proxy/update.sh:
```diff
@@ -4,68 +4,40 @@ set -o pipefail
-
+    local sha="$1"
+
+    local tarball="https://github.com/jwendell/proxy/archive/${sha}/proxy-${sha}.tar.gz"
```
should probably point to `maistra/proxy`

jwendell replied:
thanks, fixed :)

jwendell committed:
Review feedback

- Remove GN; It's not needed in community
- Fix path to download tarball in update.sh
- Add version in addition to SHA in `envoy --version`

brian-avery commented on istio-proxy/update.sh:
```diff
@@ -4,68 +4,40 @@ set -o pipefail
 done
 
-PROXY_SHA=${PROXY_SHA:-"$(grep '%global proxy_git_commit ' ${proxy_name}.spec | cut -d' ' -f3)"}
+SHA=${SHA:-"$(grep '%global git_commit ' istio-proxy.spec | cut -d' ' -f3)"}
```
Are we calling it istio-proxy.spec in product as well? The two files should match as much as possible.

jwendell replied:
no

jwendell replied:
that's the only difference - and the GN stuff now. But this GN should go away from product soon... well, that's my hope :)

brian-avery replied:
Then can we keep the variable please?

brian-avery replied:
It looks like GN is a metadata tool for Ninja: https://gn.googlesource.com/gn

jwendell replied:
Done. I've added the variable back. Switched to upper case to denote a global variable instead of a local one. With the refactoring, there are only two occurrences of it now.

brian-avery replied:
Thanks. I'm approving.

brian-avery commented on istio-proxy/update.sh:
```diff
@@ -4,68 +4,40 @@ set -o pipefail
+        echo "Already on disk, download not necessary"
+    fi
+
+    sed -i "s/%global git_commit .*/%global git_commit ${sha}/" istio-proxy.spec
```
same here.

jwendell replied:
no

brian-avery reviewed (changes requested):
A couple of minor comments, but other than that, this looks awesome. Thanks for doing this :)

brian-avery commented on istio-proxy/istio-proxy.spec:
```diff
@@ -40,44 +39,52 @@ BuildRequires:  gcc
 
-Source0:        istio-proxy.%{checksum}.tar.gz
-Source1:        build.sh
-Source2:        test.sh
```
Can the test.sh file be removed?

jwendell replied:
All files but the source .tar.gz were removed. We're invoking tests directly from .spec. Very straightforward as it should be.

brian-avery replied:
Nevermind. I see test.sh being removed now. Had to do a refresh to get it to show.

knrc reviewed (changes requested):
I'm holding this up while I work through the details with Ryan

jwendell committed:
Use PROXY_NAME variable instead of refer directly to the .spec file

dmitri-d commented on istio-proxy/istio-proxy.spec:
```diff
@@ -23,14 +23,13 @@
 %global repo            proxy
 %global provider_prefix %{provider}.%{provider_tld}/%{project}/%{repo}
```
a nit: do we need to break up github url like that? We only ever use it here...

mergify[bot] committed:
Merge branch 'maistra-1.1' into vendor-proxy
